gfx/drivers/d3d10.c

```diff
@@ -951,13 +949,9 @@ d3d10_gfx_init(const video_info_t* video,
 
    font_driver_init_osd(d3d10, false, video->is_threaded, FONT_DRIVER_RENDER_D3D10_API);
 
-   if (settings->bools.video_shader_enable)
-   {
-      const char* ext = path_get_extension(retroarch_get_shader_preset());
-
-      if (ext && !strncmp(ext, "slang", 5))
-         d3d10_gfx_set_shader(d3d10, RARCH_SHADER_SLANG, retroarch_get_shader_preset());
-   }
+   const char *shader_preset   = retroarch_get_shader_preset();
```

Comment:
C89 compliance issue. Variables need to be declared either at the top of a code block or function.